app/src/main/java/com/google/android/ground/MainActivity.kt

```diff
@@ -217,6 +217,11 @@ class MainActivity : AbstractActivity() {
   }
 
   private fun navigateTo(directions: NavDirections) {
-    navHostFragment.navController.navigate(directions)
+    navHostFragment.navController.currentDestination?.getAction(directions.actionId)?.let { action
+      ->
+      if (navHostFragment.navController.currentDestination?.id != action.destinationId) {
```

Comment:
Same here - if the screen has changed since the `navigateTo` request was enqueued, there's a design issue with the scope of the navigate requests. We should log an error and ignore.